### ISSUE TITLE
Bunch of NaNs instead of invalid DateTime

### ISSUE THREAD

**Issue by balagge** [moment/luxon#563]
Adding a long period to a DateTime object may produce invalid DateTimes, because of "overflow".
However, sometimes the `invalid` flag is not set; nevertheless, the DateTime object is unusable (contains `NaN`s)

The following example calculates the last valid DateTime objects on the timeline, and the "first" invalid ones. For the allowed range of Dates in javascript, see https://www.ecma-international.org/ecma-262/6.0/#sec-time-values-and-time-range, which says:

> –100,000,000 days to 100,000,000 days measured relative to midnight at the beginning of 01 January, 1970 UTC.

```javascript
const orig = DateTime.utc(1970,1,1,0,0,0,0); 
const N = 1E8 // max range of Dates in days
const secs = 24*60*60; // secs per day

// last valid DateTimes
const a = orig.plus({days: N}) // should be valid (it is)
console.log("a: ", a)

const b = orig.plus({seconds:N * secs}) // should be valid (it is)
console.log("b: ", b)


// first invalids
const c = orig.plus({days: N+1}) // should be invalid (it is)
console.log("c: ", c)

const d = orig.plus({seconds:N * secs+1}) // should be invalid (it is not, but a corrupt object)
console.log("d: ", d)

const e = orig.plus({seconds:(N +1)* secs}) // should also be invalid (it is not, but a corrupt object)
console.log("e: ", e)
```

**Expected behavior:** `a` and `b` should be valid, `c`, `d`, `e` should be invalid.

**Actual behavior:** 
 - `a` and `b` are valid, `c` is invalid (as expected)
 -  `d` and `e` are reported as valid, but are actually corrupt objects (error)

**Comment by neversaid** [comment]
I have added a PR for this issue. Details are written down on this as well #576

**Comment by icambron** [comment]
Ah, I hadn't read the OP carefully enough. See also #299

**Comment by icambron** [comment]
Fixed by #576
